runtime/src/bank_forks.rs

```diff
@@ -88,7 +92,9 @@ impl Index<u64> for BankForks {
 }
 
 impl BankForks {
-    pub fn new_rw_arc(root_bank: Bank) -> Arc<RwLock<Self>> {
+    pub fn new_rw_arc(mut root_bank: Bank) -> Arc<RwLock<Self>> {
+        let bank_hash_cache = Arc::new(RwLock::new(BankHashCache::default()));
+        root_bank.set_bank_hash_cache(bank_hash_cache.clone());
```

Comment:
`root_bank` is not frozen until after `BankForks` is created, so this is necessary.